Assets/Scripts/UI/Components/ATM/Timeline_Regulator.cs

```diff
@@ -0,0 +1,237 @@
+
+                float sliderValue = speedSlider.value;
+                if (sliderValue != 0f)
+                {
+                    int yearChange = Mathf.RoundToInt(sliderValue);
+                    int newYear = currentYear + yearChange;
+
+                    ScrollToYear(newYear);
+                }
+            }
+        }
+
+        void ScrollToYear(int year)
+        {
+            if (year < minYear)
+            {
+                year = minYear; // Set to lowest available year  
+            }
+            else if (year > maxYear)
+            {
+                year = maxYear; // Set to highest available year  
+            }
+
+            int index = year - minYear;
+            if (index >= 0 && index < yearInstances.Count)
+            {
+                // Set the target position to smoothly lerp to  
+                targetPosition = new Vector2((-yearInstances[index].anchoredPosition.x + 362), content.anchoredPosition.y);
+                currentYear = year; // Update the current year  
+                centerYearText.text = currentYear.ToString(); // Update the UI text  
+
+                // Adjust height, color, and text visibility  
+                for (int i = 0; i < yearInstances.Count; i++)
+                {
+                    RectTransform yearRectTransform = yearInstances[i];
+                    Transform yearText = yearRectTransform.GetChild(0);
+                    Image yearImage = yearRectTransform.GetComponent<Image>(); // Access the Image component  
+                    GameObject eventObject = yearRectTransform.GetChild(1).gameObject; // Access the event GameObject
+
+                    if (i == index) // Current year  
+                    {
+                        yearRectTransform.sizeDelta = new Vector2(yearRectTransform.sizeDelta.x, highlightedHeight);
+                        yearText.gameObject.SetActive(false); // Deactivate text  
+                        yearImage.color = highlightedColor; // Change color to #004699  
+                    }
+                    else if (i == index - 1 || i == index + 1) // Previous or next year  
+                    {
+                        yearRectTransform.sizeDelta = new Vector2(yearRectTransform.sizeDelta.x, neighborHeight);
+                        yearText.gameObject.SetActive(true); // Ensure text is active  
+                        yearImage.color = defaultColor; // Set to default color (#CCD7E4)
+                    }
+                    else // Other years  
+                    {
+                        yearRectTransform.sizeDelta = new Vector2(yearRectTransform.sizeDelta.x, defaultHeight);
+                        yearText.gameObject.SetActive(true); // Ensure text is active  
+                        yearImage.color = defaultColor; // Set to default color (#CCD7E4)
+                    }
+
+                    // Reactivate the event GameObject if the year is in the list  
+                    if (eventYears.Contains(i + minYear))
```

Comment:
var isEventYear = eventYears.Contains(i + minYear);
eventObject.SetActive(isEventYear);

Reply:
Doe ik niet omdat hij de andere eventobjecten ook uit moet zetten na de check. Als je dat netter kan maken, be my guest!

Reply:
zal ik doen